packages/swapper/src/swappers/thorchain/getThorTradeQuote/getTradeQuote.ts

```diff
@@ -53,7 +53,7 @@ export const getThorTradeQuote: GetThorTradeQuote = async ({ deps, input }) => {
 
     const sellAssetId = sellAsset.assetId
     const buyAssetId = buyAsset.assetId
-    const feeAssetId = getFeeAssetIdFromAssetId(buyAssetId)
+    const feeAssetId = adapter.getFeeAssetId()
```

Comment:
is this `feeAssetId` logic even necessary or valid? we aren't using it here and are ultimately doing this logic/check in `estimateTradeFee`.

also as future cleanup/issue, the usage of `chainId` in conjunction with `buyAsset` and `sellAsset` is super confusing (which chain id is it talking about), and not needed since we can get the applicable `chainId` of the desired asset.

Reply:
Yeah, I'm agnostic-ish to this but would tend to agree with it being unnecessary. I can see the rationale behind the early check and not going further down the call stack for no reason, but it's not like there's anything heavy happening before the `estimateTradeFee` check anyway:

https://github.com/shapeshift/lib/blob/25cc8b3bfd03225efaf2ded70a4887a8f301979f/packages/swapper/src/swappers/thorchain/utils/estimateTradeFee/estimateTradeFee.ts#L54-L55

Let's keep the check in `estimateTradeFee` to keep things simpler and remove duplication

Reply:
Cleaned up in 1cfe58a9fab2852d5e834577d79afdf0a72580cd

Reply:
Also with the early `adapter` check and considering `getFeeAssetId` is required on the chain adapter it isn't really necessary regardless. Ultimately I feel like a lot of this interface can be cleaned up to make things more clear as to what's happening, but discussions for another day. Thanks @gomesalexandre :partying_face: